### PR TITLE
Timm models

### PR DESCRIPTION
**Pull request by jonasdataloop**

### CODE REVIEW

File: adapters/timm/efficientnet/dataloop.json
Comment (RoniDataloopAi) [inline adapters/timm/efficientnet/dataloop.json:1]: "versions": {
          "dtlpy": "1.90.39"
        },
      i think it is better not to define dtlpy version ( in all manifests)

File: adapters/timm/timm_model_adapter.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+        input_size = self.configuration.get('input_size', 256)
+
+        preprocess = transforms.Compose([
+            transforms.ToPILImage(),
+            transforms.Resize((input_size, input_size)),
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+        ])
+
```
Comment (RoniDataloopAi) [inline adapters/timm/timm_model_adapter.py:85]: all models is normalized by the same mean and std? i think resent uses different values, maybe it can be an input from the configuration

File: adapters/timm/timm_model_adapter.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+import torch.nn.functional
+import numpy as np
+import torch.nn
+import logging
+import copy
+from dtlpy.utilities.dataset_generators.dataset_generator import collate_torch
+from dtlpy.utilities.dataset_generators.dataset_generator_torch import DatasetGeneratorTorch
+
+logger = logging.getLogger('MobileNetV3-adapter')
+
```
Comment (RoniDataloopAi) [inline adapters/timm/timm_model_adapter.py:20]: logger name can be changed to generic name

File: adapters/timm/timm_model_adapter.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+@dl.Package.decorators.module(name='model-adapter',
+                              description='Model Adapter for TIMM Models',
+                              init_inputs={'model_entity': dl.Model})
+class TIMMAdapter(dl.BaseModelAdapter):
+    """
+    MobileNetV3 Model adapter using Pytorch.
+    The class bind Dataloop model and model entities with model code implementation
+    """
+    def __init__(self, model_entity=None):
```
Comment (RoniDataloopAi) [inline adapters/timm/timm_model_adapter.py]: also the class description

File: adapters/timm/timm_model_adapter.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+            self.model = torch.load(model_path, map_location=self.device)
+            self.model.to(self.device)
+            self.model.eval()
+            logger.info(f"Loaded model from {model_path} successfully")
+        else:
+            # With the model from the library
+            self.model = timm.create_model(self.model_name, pretrained=True)
+            self.model.to(self.device)
+            self.model.eval()
+            logger.info(f"Loaded model from library successfully")
```
Comment (RoniDataloopAi) [inline adapters/timm/timm_model_adapter.py:57]: self.model.to(self.device)
            self.model.eval()
            
       can be outside if and else

File: adapters/timm/timm_model_adapter.py [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+            logger.warning('Failed creating shebang confusion report! Continue without...')
+
+    def convert_from_dtlpy(self, data_path, **kwargs):
+        subsets = self.model_entity.metadata.get("system", dict()).get("subsets", None)
+        if 'train' not in subsets:
+            raise ValueError(
+                'Couldnt find train set. MobileNetV3 requires train and validation set for training. '
+                'Add a train set DQL filter in the dl.Model metadata'
+            )
+        if 'validation' not in subsets:
```
Comment (RoniDataloopAi) [inline adapters/timm/timm_model_adapter.py:353]: "MobileNetV3 requires" also can be changed

File: adapters/timm/classification/efficientnet/dataloop.json [hunk 6]
```diff
@@ -0,0 +1,1194 @@
...
+  "version": "0.1.17",
+  "scope": "project",
+  "description": "EfficientNet application for image classification.",
+  "attributes": {
+    "Provider": "Dataloop",
+    "Category": "Model",
+    "Computer Vision": "Classification",
+    "Media Type": [
+      "Image"],
+    "Libraries": "Torch",
```
Comment (RoniDataloopAi) [inline adapters/timm/classification/efficientnet/dataloop.json:13]: i think attributes should be like resent and deeplab
    "Provider": "PyTorch",
    "Deployed By": "Dataloop",
    "License": "BSD-3.0",
    
    also "scope": "public"

File: adapters/timm/classification/efficientnet/dataloop.json [hunk 7]
```diff
@@ -0,0 +1,1194 @@
...
+    "gitUrl": "https://github.com/dataloop-ai-apps/torch-models.git",
+    "gitTag": "0.1.16"
+  },
+  "components": {
+    "computeConfigs": [
+      {
+        "name": "efficientnet-deploy",
+        "versions": {
+          "dtlpy": "1.90.39"
+        },
```
Comment (RoniDataloopAi) [inline adapters/timm/classification/efficientnet/dataloop.json:27]: i think you can remove 
        "versions": {
          "dtlpy": "1.90.39"
        },

File: adapters/timm/classification/efficientnet/dataloop.json [hunk 8]
```diff
@@ -0,0 +1,1194 @@
...
+      {
+        "name": "efficientnet-train-evaluate",
+        "versions": {
+          "dtlpy": "1.90.39"
+        },
+        "runtime": {
+          "podType": "gpu-t4",
+          "concurrency": 1,
+          "runnerImage": "gcr.io/viewo-g/piper/agent/gpu/torch-models:0.1.1",
+          "autoscaler": {
```
Comment (RoniDataloopAi) [inline adapters/timm/classification/efficientnet/dataloop.json:49]: if you are creating a new image so you'll need to update the image version

File: adapters/timm/classification/efficientnet/dataloop.json [hunk 9]
```diff
@@ -0,0 +1,1194 @@
...
+            "queueLength": 100
+          },
+          "executionTimeout": 36000000
+        }
+      }
+    ],
+    "modules": [
+      {
+        "name": "efficientnet-module",
+        "entryPoint": "adapters/timm/timm_model_adapter.py",
```
Comment (RoniDataloopAi) [inline adapters/timm/classification/efficientnet/dataloop.json:62]: entrypoint "entryPoint": "adapters/timm/classification/timm_model_adapter.py", ?

File: adapters/timm/classification/timm_model_adapter.py [hunk 10]
```diff
@@ -0,0 +1,357 @@
...
+        :param local_path: `str` directory path in local FileSystem
+        """
+        weights_filename = self.model_entity.configuration.get('weights_filename')
+        model_path = str(os.path.join(local_path, weights_filename))
+        if weights_filename and os.path.exists(model_path):
+            # With the model entity and weights from the platform
+
+            logger.info(f"Loading a model from {model_path}")
+            self.model = torch.load(model_path, map_location=self.device)
+            logger.info(f"Loaded model from {model_path} successfully")
```
Comment (RoniDataloopAi) [inline adapters/timm/classification/timm_model_adapter.py:52]: it isn't possible to create using timm with checkpoint path?

Comment (RoniDataloopAi) [inline adapters/timm/classification/timm_model_adapter.py:52]: it isn't possible to create using timm with checkpoint path?
